modules/api/src/main/java/org/eclipse/xpanse/api/OrchestratorApi.java

```diff
@@ -44,78 +51,205 @@ public class OrchestratorApi {
 
     private final OrchestratorService orchestratorService;
 
+    private final RegisterService registerService;
+
     @Autowired
-    public OrchestratorApi(OrchestratorService orchestratorService) {
+    public OrchestratorApi(OrchestratorService orchestratorService,
+            RegisterService registerService) {
         this.orchestratorService = orchestratorService;
+        this.registerService = registerService;
     }
 
     /**
-     * Register new managed service.
+     * List cloud service providers.
      *
-     * @param ocl object of managed service.
      * @return response
      */
-    @Tag(name = "Service Vendor",
-            description = "APIs to publish the managed services.")
+    @Tag(name = "Service Provider or Service Vendor",
+            description = "APIs to manage register services.")
+    @Operation(description = "List cloud service providers.")
+    @GetMapping(value = "/register/providers",
+            produces = MediaType.APPLICATION_JSON_VALUE)
+    @ResponseStatus(HttpStatus.OK)
+    @Transactional
+    public Response providers() {
```

Comment:
we already have the providers as an enum and any client can/must simply use  that directly. I think we dont need this extra service.

Reply:
By calling this service. When we change the list of providers,  we only need to modify the enum on the server side, not the client side.

Reply:
Ok. in this case. the issue is applicable to all enums. Not just this one. In my opinion, the dependent clients must do an update if needed.

Reply:
Now the UI client shows baseed on return data. I remove the API